File: src/main/java/com/example/firstdemo/pojo/Property.java
package com.example.firstdemo.pojo;

import com.fasterxml.jackson.annotation.JsonFormat;
import org.springframework.format.annotation.DateTimeFormat;

import java.io.Serializable;
import java.math.BigDecimal;
import java.util.Date;

/**
 * @description: 知产管理
 *
 * @author: Logan
 * @create: 2019/06/20
 *
 */
public class Property implements Serializable {
    private static final long serialVersionUID = 1L;

    private Long propertyId;

    private Long userId;

    private Integer propertyTypeId;

    private String propertyName;

    private String equipmentNo;

    private BigDecimal propertyPrice;

    private Integer count;

    @DateTimeFormat(pattern="yyyy-MM-dd HH:mm:ss")//页面写入数据库时格式化
    @JsonFormat(pattern = "yyyy-MM-dd HH:mm:ss")//数据库导出页面时json格式化
    private Date createTime;

    @DateTimeFormat(pattern="yyyy-MM-dd HH:mm:ss")//页面写入数据库时格式化
    @JsonFormat(pattern = "yyyy-MM-dd HH:mm:ss")//数据库导出页面时json格式化
    private Date updateTime;

    private String description;

    private String registrant;


    public Long getPropertyId() {
        return propertyId;
    }

    public void setPropertyId(Long propertyId) {
        this.propertyId = propertyId;
    }

    public Long getUserId() {
        return userId;
    }

    public void setUserId(Long userId) {
        this.userId = userId;
    }

    public Integer getPropertyTypeId() {
        return propertyTypeId;
    }

    public void setPropertyTypeId(Integer propertyTypeId) {
        this.propertyTypeId = propertyTypeId;
    }

    public String getPropertyName() {
        return propertyName;
    }

    public void setPropertyName(String propertyName) {
        this.propertyName = propertyName;
    }

    public String getEquipmentNo() {
        return equipmentNo;
    }

    public void setEquipmentNo(String equipmentNo) {
        this.equipmentNo = equipmentNo;
    }

    public BigDecimal getPropertyPrice() {
        return propertyPrice;
    }

    public void setPropertyPrice(BigDecimal propertyPrice) {
        this.propertyPrice = propertyPrice;
    }

    public Integer getCount() {
        return count;
    }

    public void setCount(Integer count) {
        this.count = count;
    }

    public Date getCreateTime() {
        return createTime;
    }

    public void setCreateTime(Date createTime) {
        this.createTime = createTime;
    }

    public Date getUpdateTime() {
        return updateTime;
    }

    public void setUpdateTime(Date updateTime) {
        this.updateTime = updateTime;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public String getRegistrant() {
        return registrant;
    }

    public void setRegistrant(String registrant) {
        this.registrant = registrant;
    }
}

File: src/main/resources/static/app/model/Property.js
Ext.define("App.model.Property", {
    extend: "Ext.data.Model",

    idProperty: 'propertyId',

    fields: [
        { name: 'propertyId', type: 'int'},
        { name: 'userId', type: 'int',
            mapping: {

            }},
        { name: 'propertyTypeId', type: 'int' },
        { name: 'propertyName', type: 'string' },
        { name: 'equipmentNo', type: 'string' },
        { name: 'propertyPrice', type: 'auto' },
        { name: 'count', type: 'int' },
        { name: 'description', type: 'string'},
        { name: 'registrant', type: 'string'},
        { name: 'createTime', type: 'date',
            convert: function(value){
                if(!Ext.isEmpty(value)) {
                    return Ext.util.Format.date(value, 'Y-m-d H:i:s');
                }　else {
                    return value;
                }
            }},
        { name: 'updateTime', type: 'date',
            convert: function(value){
                if(!Ext.isEmpty(value)) {
                    return Ext.util.Format.date(value, 'Y-m-d H:m:s');
                }　else {
                    return value;
                }
            }},
        { name: 'propertyTypeName', type: 'string',
            convert: function(value, red){
                if(!Ext.isEmpty(red.get('propertyType'))) {
                    value = red.get('propertyType').propertyTypeName;
                }
                return value;
            }},
        { name: 'userName', type: 'string',
            convert: function(value, red){
                if(!Ext.isEmpty(red.get('users'))) {
                    value = red.get('users').userName;
                }
                return value;
            }}
    ]
});